qiskit/providers/aer/library/save_expectation_value.py

```diff
@@ -0,0 +1,246 @@
+        super().__init__('save_expval',
+                         key,
+                         operator.num_qubits,
+                         conditional=conditional,
+                         pershot=pershot,
+                         unnormalized=unnormalized,
+                         params=params)
+
+
+class SaveExpectationValueVariance(SaveAverageData):
+    """Save expectation value and variance of an operator."""
+    def __init__(self,
+                 key,
+                 operator,
+                 unnormalized=False,
+                 conditional=False,
+                 pershot=False):
+        r"""Instruction to save the expectation value and variance of a Hermitian operator.
+
+        The expectation value of a Hermitian operator :math:`H` for a
+        simulator in quantum state :math`\rho`is given by
+        :math:`\langle H\rangle = \mbox{Tr}[H.\rho]`. The variance is given by
+        :math:`\sigma^2 = \langle H^2 \rangle - \langle H \rangle>^2`.
+
+        Args:
+            key (str): the key for retrieving saved data from results.
+            operator (Pauli or SparsePauliOp or Operator): a Hermitian operator.
+            unnormalized (bool): If True return save the unnormalized accumulated
+                                 or conditional accumulated expectation value
+                                 over all shot [Default: False].
+            pershot (bool): if True save a list of expectation values for each shot
+                            of the simulation rather than the average over
+                            all shots [Default: False].
+            conditional (bool): if True save the average or pershot data
+                                conditional on the current classical register
+                                values [Default: False].
+
+        Raises:
+            ExtensionError: if the input operator is invalid or not Hermitian.
+
+        .. note:
+            This instruction can be directly appended to a circuit using
+            the :func:`save_expectation_value` circuit method.
+        """
+        # Convert O to SparsePauliOp representation
+        if isinstance(operator, Pauli):
+            operator = SparsePauliOp(operator)
+        elif not isinstance(operator, SparsePauliOp):
+            operator = SparsePauliOp.from_operator(Operator(operator))
+        if not isinstance(operator, SparsePauliOp):
+            raise ExtensionError("Invalid input operator")
+        if not allclose(operator.coeffs.imag, 0):
+            raise ExtensionError("Input operator is not Hermitian.")
+        params = _expval_params(operator, variance=True)
+        super().__init__('save_expval_var',
+                         key,
+                         operator.num_qubits,
+                         conditional=conditional,
+                         pershot=pershot,
+                         unnormalized=unnormalized,
+                         params=params)
```

Comment:
This is almost the same as the code in `SaveExpectationValue`, maybe we can make a common parent and pass to its constructor the `name`: `save_expval` or `save_expval_var` and if `variance` is `False` or `True`.

Reply:
We could but I think making a common base class adds more lines of code than it removes from the copy pasted parts.